eo-parser/src/main/java/org/eolang/parser/ObjectNameFailure.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2016-2025 Objectionary.com
+ * SPDX-License-Identifier: MIT
+ */
+package org.eolang.parser;
+
+import java.nio.file.Path;
+import java.util.function.Supplier;
+import org.cactoos.Proc;
+import org.cactoos.proc.UncheckedProc;
+
+/**
+ * Object name failure.
+ * If {@link ObjectName} throw an exception, this object catches it, and executes supplied
+ * procedure. This object is supposed to be used together with {@link ObjectName}.
+ *
+ * @since 0.56.5
+ */
+public final class ObjectNameFailure implements Supplier<String> {
```

Comment:
@h1alexbel `ObjectNameFailure` sounds confusing. It's not more object name, it's failure... Let's rename it to something like `ObjectNameSafe` or `ObjectNameCatched`